src/pages/Xudt/HolderAllocation.tsx

```diff
@@ -6,56 +6,61 @@ import CloseIcon from '../../assets/modal_close.png'
 import SimpleButton from '../../components/SimpleButton'
 
 const HolderAllocation = ({
-  allocation,
+  ckbHolderAmount,
+  btcHolderAmount,
+  lockHoderAmount,
   onClose,
 }: {
-  allocation: Record<string, number>
+  ckbHolderAmount: string
+  btcHolderAmount: string
+  lockHoderAmount?: {
+    name: string
+    holderCount: string
+    codeHash: string
+  }[]
   onClose: MouseEventHandler<HTMLDivElement>
 }) => {
   const [t] = useTranslation()
-  const total = Object.values(allocation).reduce((acc, cur) => acc + cur, 0)
-  const btc = allocation.BTC
-  const ckb = total - btc
   return (
     <div className={styles.holderAllocationContainer}>
       <div className={styles.holderAllocationContent}>
         <h2>{t('xudt.holder_allocation')}</h2>
         <p>
           {t('xudt.holder_allocation_description', {
-            ckb: localeNumberString(ckb),
-            btc: localeNumberString(btc),
+            ckb: ckbHolderAmount,
+            btc: localeNumberString(btcHolderAmount),
           })}
         </p>
-        <div className={styles.table}>
-          <table>
-            <thead>
-              <tr>
-                <td>
-                  <div>{t('xudt.category')}</div>
-                </td>
-                <td>
-                  <div>{t('xudt.count')}</div>
-                </td>
-              </tr>
-            </thead>
-            <tbody>
-              {Object.entries(allocation)
-                .sort((a, b) => b[1] - a[1])
-                .map(([label, count]) => {
-                  return (
+        {lockHoderAmount && (
+          <div className={styles.table}>
+            <table>
+              <thead>
+                <tr>
+                  <td>
+                    <div>{t('xudt.lock_hash')}</div>
+                  </td>
+                  <td>
+                    <div>{t('xudt.count')}</div>
+                  </td>
+                </tr>
+              </thead>
+              <tbody>
+                {lockHoderAmount
+                  .sort((a, b) => +b.holderCount - +a.holderCount)
+                  .map(amount => (
                     <tr>
                       <td>
-                        <div>{label}</div>
+                        <div>{amount.name ?? `#${amount.codeHash.slice(2, 6)}`}</div>
```

Comment:
The label should be matched on the frontend because the script set on the frontend is larger than that on the backend.

And generally a hash will be displayed with ellipsis in the middle

Reply:
To achieve this, the whole script should be in response. @rabbitz